src/Symfony/Component/Yaml/Command/LintCommand.php

```diff
@@ -137,40 +143,75 @@ private function display(SymfonyStyle $io, array $files): int
                 return $this->displayTxt($io, $files);
             case 'json':
                 return $this->displayJson($io, $files);
+            case 'github':
+                return $this->displayTxt($io, $files, true);
             default:
                 throw new InvalidArgumentException(sprintf('The format "%s" is not supported.', $this->format));
         }
     }
 
-    private function displayTxt(SymfonyStyle $io, array $filesInfo): int
+    private function displayTxt(SymfonyStyle $io, array $filesInfo, bool $errorAsGithubAnnotations = false): int
     {
         $countFiles = \count($filesInfo);
         $erroredFiles = 0;
         $suggestTagOption = false;
 
+        $buffer = [];
         foreach ($filesInfo as $info) {
             if ($info['valid'] && $this->displayCorrectFiles) {
                 $io->comment('<info>OK</info>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
             } elseif (!$info['valid']) {
                 ++$erroredFiles;
-                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
-                $io->text(sprintf('<error> >> %s</error>', $info['message']));
 
                 if (false !== strpos($info['message'], 'PARSE_CUSTOM_TAGS')) {
                     $suggestTagOption = true;
                 }
+
+                if ($errorAsGithubAnnotations) {
+                    $buffer[] = $this->formatErrorAsGithubAnnotation($info);
+                    continue;
```

Comment:
Errors should still be rendered using the normal text format in addition to the GitHub annotation. Actions have a limit on the number of annotations they can report. So if you have lots of errors, you won't see them all in annotations.
If we keep rendering the normal textual report, you can go to the `checks` tab to see the full report for such case.

So I would remove the `continue`

Reply:
That'll duplicate errors reports on the output, though.
Do you know what the limit is?

Reply:
The old API v3 doc mentions:

> The Checks API limits the number of annotations to a maximum of 50 per API request. To create more than 50 annotations, you have to make multiple requests to the Update a check run endpoint.

https://developer.github.com/v3/checks/runs/#:~:text=Annotations%20are%20visible%20in%20GitHub's,Update%20a%20check%20run%20endpoint.

But no info about the max shown annotations.

https://github.community/t/maximum-number-of-annotations-that-can-be-created-using-github-actions-logging-commands/16991/3

states:

> It is by designed that 10 annotation created by Github Actions logging command.

>Summarize the limit of annotation as below:
>
>- 10 warning annotations and 10 error annotations per step
>- 50 annotations per job (sum of annotations from all the steps)
>- 50 annotations per run (separate from the job annotations, these annotations aren’t created by users)

Reply:
@ogizanagi all the special formats are at the end of the output of the command though, meaning you can stop scrolling when you read the github annotations.

IIRC, the limit is 50 annotations (not sure whether it is per step or per job, but the command would be a single step anyway)

Reply:
> all the special formats are at the end of the output of the command though, meaning you can stop scrolling when you read the github annotations.

I see. Didn't realized that. So let's drop the `continue` indeed.